libraries/autoload.rb

```diff
@@ -7,6 +7,7 @@
 # NOTE: if the gem is installed but on a different version, we simply don't care: the cookbook version wins.
 #       If another gem depends on a different version being there, rubygems will correctly throw the error.
 if Gem.loaded_specs["compat_resource"]
+  Chef.log_deprecation "using compat_resource as a gem is deprecated and will be removed in a future version of this cookbook"
```

Comment:
since we loaded up as a gem we should have Chef.log_deprecation already monkeypatched in -- i suppose someone might have some brutally old version of the gem loaded up that doesn't support Chef.log_deprecation but i'm pretty sure we've supported that from the start or nearly from the start.